File: Pieces/Knight.mjs
// import Square from "../Square.mjs";
import Piece from "./Piece.mjs";

export default class Knight extends Piece {
  constructor(isWhite) {
    super(isWhite);
  }

  pieceSymbol = () => {
    return this.isWhite ? "♞" : "♘";
  };

  move = (initialSquare, destinationSquare) => {
    let dx = Math.abs(destinationSquare.x - initialSquare.x);
    let dy = Math.abs(destinationSquare.y - initialSquare.y);
    if (dx * dy !== 2) return false;

    switch (this.isWhite) {
      case true: {
        if (destinationSquare.piece?.isWhite) return false;

        break;
      }

      case false: {
        if (!destinationSquare.piece?.isWhite) return false;

        break;
      }
    }

    return true;
  };
}

File: Pieces/py/Knight.py
import Piece

class Knight(Piece):
    def __init__(self, is_white):
        super().__init__(is_white)

    def move(self, initial_square, destination_square):
        if destination_square.piece and destination_square.piece.is_white == self.is_white:
            return False
        
        dx = abs(initial_square.x - destination_square.x)
        dy = abs(initial_square.y - destination_square.y)

        if dx * dy != 2:
            return False
        
        return True